traefik.go

```diff
@@ -1,16 +1,130 @@
 package main
 
 import (
+	"encoding/json"
+	log "github.com/Sirupsen/logrus"
+	"github.com/containous/flaeg"
+	"github.com/containous/staert"
+	"github.com/containous/traefik/middlewares"
+	"github.com/containous/traefik/provider"
 	fmtlog "log"
+	"net/http"
 	"os"
+	"reflect"
 	"runtime"
+	"strings"
 )
 
 func main() {
 	runtime.GOMAXPROCS(runtime.NumCPU())
-	if err := traefikCmd.Execute(); err != nil {
+
+	//traefik config inits
+	traefikConfiguration := NewTraefikConfiguration()
+	traefikPointersConfiguration := NewTraefikPointersConfiguration()
```

Comment:
`flaeg` loads values for args (or flags) into the configuration structure. 
Non pointers fields have can be initialized before `flaeg` process. Func `NewTraefikConfiguration` is in charge of this initialization. 
It's a bit trickier for pointers fields, their default value have to be stored in another structure: here it's `traefikPointersConfiguration`